books.php

```diff
@@ -1,36 +1,99 @@
 <?php
-class Books
+require_once('DB.php');
+function auth($login, $password)
 {
-    private static $instance = null;
-    private function __construct()
-    {
-        $this->link = new PDO('mysql: host=localhost;dbname=library_bd', "root", "root");
+    $stmt = DB::getInstance()->link()->prepare("SELECT id FROM users WHERE 
+        BINARY login = :login AND BINARY password = :password");
+    $stmt->execute([':login' => $login, ':password' => $password]);
+    if ($stmt->rowCount() > 0) {
+        return true;
     }
-    private function __clone()
-    { }
-    public static function getInstance()
-    {
-        if (self::$instance === null) {
-            self::$instance = new Books;
-        }
-        return self::$instance;
+    return false;
+}
+
+function add($data)
+{
+    // необходимые ключи для таблицы
+    $arrayTable = array_flip(['name', 'genre', 'pages']);
+    // оставляем только необходимые ключи
+    $book = array_intersect_key($data, $arrayTable);
+    // каких ключей не хватает
+    $dif = array_diff_key($arrayTable, $book);
+    // если ключей меньше, чем нужно - возвращаем ошибку
+    if (count($book) != count($arrayTable)) {
+        return ['data' => $data, 'message' => 'Needed fields are ' .
+            implode(',', array_keys($dif)), 'status' => 500];
     }
-    public function allBooks()
-    {
-        $query = "SELECT * FROM book";
-        $resQuery = $this->link->query($query);
-        echo json_encode($resQuery->fetchAll(PDO::FETCH_ASSOC));
+    $stmt = DB::getInstance()->link()->prepare(
+        "INSERT INTO `book`(`name`, `genre`, `pages`) 
+                        VALUES (:name, :genre, :pages);"
+    );
+    $stmt->bindParam(":name", $book['name'], PDO::PARAM_STR);
+    $stmt->bindParam(":genre", $book['genre'], PDO::PARAM_STR);
+    $stmt->bindParam(":pages", $book['pages'], PDO::PARAM_INT);
+    $stmt->execute();
+    $i = $stmt->rowCount();
+    $id = DB::getInstance()->link()->lastInsertId();
+    return ['data' => 'id=>' . $id, 'message' => $i . ' added books', 'status' => 200];
+}
+
+function update($data, $id)
+{
+    $arrayTable = array_flip(['name', 'genre', 'pages']);
+    $book = array_intersect_key($data, $arrayTable);
+    $set = '';
+    foreach ($book as $key => $value) {
+        $set .= "`$key`=:$key,";
     }
-    public function booksId($id)
-    {
-        $query = "SELECT * FROM book WHERE id = $id";
-        $resQuery = $this->link->query($query);
-        echo json_encode($resQuery->fetch(PDO::FETCH_ASSOC));
+    $set = substr($set, 0, -1);

```

Comment:
to avoid this logic in previous foreach work with array instead of string.
like:
    $setArr = [];
    foreach ($book as $key => $value) {
        $setArr[] = "`$key`=:$key";
    }
    $set = implode(', ', $setArr);